logger_test.go

```diff
@@ -930,7 +930,18 @@ func TestConfigFromEnvOk(t *testing.T) {
 	}
 }
 
-// io.Discard but specially known to by logger optimizations for instance.
+func TestInvalidFile(t *testing.T) {
+	prev := jWriter.w
+	invalidFile := os.NewFile(^uintptr(0), "invalid-file")
+	jWriter.w = invalidFile
+	b := ConsoleLogging()
+	jWriter.w = prev
+	if b {
+		t.Errorf("expected not to be console logging")
+	}
+}
+
+// like io.Discard except io.Discard is checked for by logger optimizations and we want to avoid that.
```

Comment:
```suggestion
// like io.Discard except io.Discard is checked for by logger optimizations and we want to avoid that.
// e.g. https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/log/log.go;l=84
```